augur/filter/weights_file.py

```diff
@@ -0,0 +1,37 @@
+import csv
+import pandas as pd
+from textwrap import dedent
+from augur.errors import AugurError
+
+
+WEIGHTS_COLUMN = 'weight'
+
+
+class InvalidWeightsFile(AugurError):
+    def __init__(self, file, error_message):
+        super().__init__(f"Bad weights file {file!r}.\n{error_message}")
+
+
+def read_weights_file(weights_file):
+    weights = pd.read_csv(weights_file, delimiter='\t')
```

Comment:
per [Tom's recent PR](https://github.com/nextstrain/augur/pull/1561) maybe set `newline` here?

```suggestion
    weights = pd.read_csv(weights_file, newline='', delimiter='\t')
```

Reply:
@genehack I thought Tom's recommendation was focused on the built-in `csv` module and not the pandas `read_csv` function which doesn't have a `newline` argument? [pandas seems to open files internally](https://github.com/pandas-dev/pandas/blob/7ee10913d7192a285383ceec9539191b784c1ed4/pandas/io/formats/format.py#L1048) with `newline=""`.

Reply:
> [pandas seems to open files internally](https://github.com/pandas-dev/pandas/blob/7ee10913d7192a285383ceec9539191b784c1ed4/pandas/io/formats/format.py#L1048) with `newline=""`.

🤦 you are correct, i over-indexed on `csv`...